src/views/EventLog.vue

```diff
@@ -25,41 +44,61 @@
     const formatDate = new Intl.DateTimeFormat(undefined, options).format;
 
     export default {
-        name: "EventLog",
-        data() {
-            return {
-                search: '',
-                loadingStatus: false,
-                events: [],
-                headers: [
-                    { text: 'Time', value: 'readableDate'},
-                    { text: 'User', value: 'details.username' },
-                    { text: 'Event type', value: 'type' },
-                    { text: 'Application', value: 'clientId' },
-                ],
-            }
-        },
+      name: "EventLog",
+      data() {
+        return {
+          searchUserId: '',
+          searchApplication: '',
+          searchDateFrom: '',
+          searchDateTo: '',
+          filterEvents: '',
+          loadingStatus: false,
+          events: [],
+          headers: [
+            {text: 'Time', value: 'readableDate'},
+            {text: 'User', value: 'userId'},
+            {text: 'Event type', value: 'type'},
+            {text: 'Application', value: 'clientId'},
+          ],
+        }
+      },
 
-        created() {
-            this.getEvents();
-        },
+      created() {
+        this.getAllEvents();
+      },
 
-        methods: {
-            getEvents: async function () {
-              this.loadingStatus = true;
-              try {
-                let promise = await EventsRepository.getEvents();
-                this.events = promise.data;
-                for (let e of this.events) {
-                  e.readableDate = formatDate(e.time);
-                  if (!e.details) {
-                    e.details = 'no details';
-                  }
-                }
-              } finally {
-                this.loadingStatus = false;
+      methods: {
+        searchEvents: function () {
+          const params = new URLSearchParams();
+          [
+            {name: 'user', value: this.searchUserId},
+            {name: 'client', value: this.searchApplication},
+            {name: 'dateFrom', value: this.searchDateFrom},
+            {name: 'dateTo', value: this.searchDateTo},
+          ].map(param => {
+            if (param.value) params.append(param.name, param.value);
+          });
```

Comment:
This I argue is nice. It's cleaner than four `if` statements, and it's easy to add more search fields.